app/webpacker/components/EventsTable/index.jsx

```diff
@@ -64,51 +54,56 @@ export default function EventsTable({ competitionInfo, wcifEvents }) {
         </TableHeader>
 
         <TableBody>
-          {wcifEvents.map((event) => event.rounds.map((round, i) => (
-            <TableRow key={round.id}>
-              {i === 0 && (
-                <TableCell rowSpan={event.rounds.length}>
-                  {events.byId[event.id].name}
-                </TableCell>
-              )}
-              <TableCell>
-                {determineRoundLabel(i + 1, event.rounds.length, round)}
-              </TableCell>
-              <TableCell>
-                {round.cutoff && `${formats.byId[round.cutoff.numberOfAttempts].shortName} / `}
-                {formats.byId[round.format].shortName}
-              </TableCell>
-              <TableCell>
-                {timeLimitToString(round, wcifEvents)}
-                {round.timeLimit !== null && (
-                  <>
-                    {round.timeLimit.cumulativeRoundIds.length === 1 && (
-                      <a href="#cumulative-time-limit">*</a>
-                    )}
-                    {round.timeLimit.cumulativeRoundIds.length > 1 && (
-                      <a href="#cumulative-across-rounds-time-limit">**</a>
-                    )}
-                  </>
+          {wcifEvents.map((event) => event.rounds.map((round, i) => {
+            const roundTypeId = getRoundTypeId(i + 1, event.rounds.length, Boolean(round.cutoff));
+            const isH2hFinal = determineH2hfinal(roundTypeId, round.id);
+
+            return (
+              <TableRow key={round.id}>
+                {i === 0 && (
+                  <TableCell rowSpan={event.rounds.length}>
+                    {events.byId[event.id].name}
+                  </TableCell>
                 )}
-              </TableCell>
-              {competitionInfo['uses_cutoff?'] && (
                 <TableCell>
-                  {round.cutoff
-                    && cutoffToString(round)}
+                  <TableCell>{I18n.t(`rounds.${roundTypeId}.cell_name`)}</TableCell>
```

Comment:
In my diff it looks like you've wrapped a TableCell within a TableCell, which does not make much sense. What is going on here?

Reply:
<img width="639" height="138" alt="Image" src="https://github.com/user-attachments/assets/af2f0185-b4bf-4cb0-9a0e-4861eaf7a60e" />